src/client/providers/confirmation-provider.ts

```diff
@@ -0,0 +1,192 @@
+import { UserConfirmationProvider, ToolExecutionDetails, SettingsProvider, UserSettings } from '../types.js';
+import { logger } from '../../utils/logger.js';
+import * as readline from 'readline';
+import { FileSettingsProvider } from './file-settings-provider.js';
+import chalk from 'chalk';
+import boxen from 'boxen';
+
+/**
+ * Default implementation of UserConfirmationProvider
+ * Automatically approves tools in the allowedTools set,
+ * and prompts for confirmation for other tools using logger's styling
+ */
+export class CLIConfirmationProvider implements UserConfirmationProvider {
+    public allowedTools: Set<string>;
+    private rl: readline.Interface | null = null;
+    public settings: Promise<UserSettings>;
+
+    constructor(allowedTools?: Set<string>, settingsProvider?: SettingsProvider) {
+        this.allowedTools = allowedTools ?? new Set();
+        this.settings = new FileSettingsProvider().getUserSettings();
```

Comment:
this line would then change to be

```
this.settingsProvider = settingsProvider ?? new FileSettingsProvider()
```

and then later places, wherever we check for settings